odl/space/base_tensors.py

```diff
@@ -91,18 +92,18 @@ def __init__(self, shape, dtype):
         self.__shape = dtype.shape + shape
         self.__dtype = dtype.base
 
-        if is_real_dtype(self.dtype):
+        if is_real_dtype(self.__dtype):
```

Comment:
Well that was an actual issue. I don't recall all the details, but it went wrong in one of the subclasses, probably `DiscreteLp`. The issue is that `self.__dtype` refers to the thing we just set, whereas `self.dtype` refers to whatever the subclass does, and that went wrong at some point.

I think we need to be careful in general with using the public property in `__init__`, maybe it would be better to consequently always use the private value in `__init__` and the public one only later when the instance is "finished".

Reply:
I made an issue about it: #1260